var/spack/repos/builtin/packages/veloc/package.py

```diff
@@ -17,7 +17,8 @@ class Veloc(CMakePackage):
     tags = ['ecp']
 
     version('master', branch='master')
-    version('1.1',    sha256='2bbdacf3e0ce4e7c9e360874d8d85b405525bdc7bd992bdb1f1ba49218072160', preferred=True)
+    version('1.4', tag='veloc-1.4', preferred=True)
+    version('1.1',    sha256='2bbdacf3e0ce4e7c9e360874d8d85b405525bdc7bd992bdb1f1ba49218072160')
     version('1.0',    sha256='d594b73d6549a61fce8e67b8984a17cebc3e766fc520ed1636ae3683cdde77cb')
     version('1.0rc1', sha256='81686ca0994a22475911d38d21c7c74b64ffef4ca872fd01f76d155c5124b0bc')
 
```

Comment:
This boost dependency seems overconstrained. Are all of those variants required to be off? If not, I would remove them and only specify the hard requirements.

Reply:
I don't know if those are required. I will try and ping the original author of the package. I just want to use version 1.4

Reply:
@gonsie Do you know if the variants specified for `veloc`'s `boost` dependency are strictly required?

Reply:
The goal is to build boost as a header-only dependency. These restrictions on boost greatly speed up the overall build time.

Reply:
What do you think Adam? Can we merge this as is for now, just adding version 1.4, and potentially do a separate PR down the road that might give the user the option of relying on a header only boost vs using a less-constrained boost?

Reply:
I don't use boost, but it would be nice if packages only enabled/disabled the variants they needed so that you don't have to build multiple copies of boost. The less constrained things are, the fewer rebuilds. Users can always change variant defaults in their packages.yaml without forcing those constraints on everyone else.

Reply:
@adamjstewart is changing variant defaults something new in spack? I haven’t heard of that before. Is there a way that we say that we are compatible with any variant of boost?

Reply:
I think the idea is to only specify the variants which are strictly required in a `depends_on` clause. For instance, if the `depends_on` just says `boost`, then a user is able to build `precice ^boost+variant1+variant2~variant3` whereas if you explicitly say `depends_on` `boost~variant1~variant2~variant3` then an end user absolutely has to build `precice` against a version of boost with all the variants turned off.

Reply:
Like @eugeneswalker said, you can override the default values on the command-line or via `packages.yaml`: https://spack.readthedocs.io/en/latest/build_settings.html#concretization-preferences

`depends_on` lists hard requirements, not preferences. That's why I like to keep it as relaxed as possible.

Reply:
Ah, that is my misunderstanding then. You can go ahead and relax this requirement to just boost then.

Reply:
Will do this tomorrow and ping for review then. Thank you!

Reply:
I've unconstrained the `boost` dependency and done a sample build to verify that it works.